### ISSUE TITLE
Mark SyncTriggers [RequiresRunningHost]

### ISSUE THREAD

**Issue by mathewc** [Azure/azure-functions-host#9904]
The SyncTriggers admin operation is a WebHost level API that currently doesn't require a running ScriptHost (see API [here](https://github.com/Azure/azure-functions-host/blob/1c0a2ef7b44e690fb70dd8d125f6b3788cf198f0/src/WebJobs.Script.WebHost/Controllers/HostController.cs#L389) - it's not marked `[RequiresRunningHost]`). Historically when we only had file based function.json indexing that hasn't been a problem, because the metadata provider didn't requires a running host - it just indexed function directories.

With the recent addition of worker indexing, we need to verify that those assumptions still hold, and ensure that the SyncTriggers behavior is correct in cases where the worker can't start. FunctionsSyncManager uses the WebHost metadata provider [here](https://github.com/Azure/azure-functions-host/blob/1c0a2ef7b44e690fb70dd8d125f6b3788cf198f0/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L309) to get metadata. If this metadata provider can't return reliable information (e.g. worker can't start) it needs to hard fail - not return empty, since the latter could result in oscillating sync operations in the case of transient failures.

Similarly, Functions also supports custom [IFunctionProvider](https://github.com/Azure/azure-functions-host/blob/d2157eaf4623c2f7aa14b8d85b10bb766655d9be/src/WebJobs.Script.Abstractions/Description/IFunctionProvider.cs#L9) implementations that can be provided by extensions when they're registered. Metadata from these providers *requires* a running host, since they're only added when the inner host has started.

In summary, we need to spend time exploring these worker indexing scenarios to see if things are behaving the way we expect. However, I do think we should just mark the operation RequiresRunningHost.
